lib/v2/notefolio/search.js

```diff
@@ -0,0 +1,158 @@
+        keywords: 'crafts',
+        category: 'portfolio.category.jessicaking',
+    },
+    {
+        keywords: 'fine-art',
+        category: 'portfolio.category.johnmedina',
+    },
+];
+
+/**
+ * @param category 分类
+ * @param order 排序 [like, pick, published]
+ * @param time 时间范围 ['one-day', 'week', 'month', 'three-month']
+ * @param query 搜索词
+ * @query limit 分页，默认20
+ */
+module.exports = async (ctx) => {
+    // 分类、排序、时间范围、搜索词
+    const { category = 'all', order = 'like', time = 'all', query = '' } = ctx.params;
+    //  分页
+    const { limit } = ctx.query;
+
+    // 请求链接
+    let searchUrl = `https://api.stunning.kr/api/v1/dantats/portfolio?state=Public&limit=${limit ? parseInt(limit, 10) : 20}&search=${query}&orderBy=${order}`;
+    // 分类
+    const index = (Number(category) || 1) - 1;
+    if (!category && index >= 0 && categoryMap[index]) {
+        searchUrl += `&category=${categoryMap[index].category}`;
+    }
+    // 时间范围
+    if (time !== 'all' && ['one-day', 'week', 'month', 'three-month'].includes(time)) {
+        let startTime = '';
+        const endTime = dayjs().endOf('d').format('YYYY-MM-DDTHH:mm:ss.SSS');
+
+        // 过去24小时-day 最近一周-week 最近一个月-month 最近三个月three-month
+        switch (time) {
+            case 'one-day': {
+                startTime = dayjs().subtract(1, 'd').format('YYYY-MM-DDTHH:mm:ss.SSS');
+                break;
+            }
+            case 'week': {
+                startTime = dayjs().subtract(7, 'd').startOf('d').format('YYYY-MM-DDTHH:mm:ss.SSS');
+                break;
+            }
+            case 'month': {
+                startTime = dayjs().subtract(30, 'd').startOf('d').format('YYYY-MM-DDTHH:mm:ss.SSS');
+                break;
+            }
+            case 'three-month': {
+                startTime = dayjs().subtract(90, 'd').startOf('d').format('YYYY-MM-DDTHH:mm:ss.SSS');
+                break;
+            }
+        }
+        searchUrl += `&publishedAt=${startTime}Z&publishedAt=${endTime}Z`;
+    }
+
+    // 发送 HTTP GET 请求到 API 并解构返回的数据对象
+    const { data } = await got(searchUrl, {
+        headers: {
+            Origin: 'https://notefolio.net',
+        },
+    });
+
+    // 从 API 响应中提取相关数据
+    const items =
+        data?.resultData.map((item) => {
+            const { id, title, user, createdAt, categories = [], contents = [] } = item;
+
+            const description = contents
+                .map((m) => {
+                    const { type } = m;
+                    let text = '';
+
+                    if (type === 'Image') {
+                        text = m.files.map((f) => `<img src='${f.url}' alt style="max-width: 100%; height: auto;">`);
+                    } else if (type === 'Text') {
+                        text = m.content;
+                    } else if (type === 'Video') {
+                        text = `<iframe src='${m.url}'></iframe>`;
+                    }
+
+                    return text;
+                })
+                .join(' ');
```

Comment:
Please use `art-template` for custom HTML content.
docs: https://docs.rsshub.app/joinus/script-standard.html